runtime/contract_update_validation.go

```diff
@@ -214,6 +216,43 @@ func (validator *ContractUpdateValidator) checkNestedDeclarations(
 
 		validator.checkDeclarationUpdatability(oldNestedDecl, newNestedDecl)
 	}
+
+	// Check enum-cases, if theres any.
+	validator.checkEnumCases(oldDeclaration, newDeclaration)
+}
+
+// checkEnumCases validates updating enum cases. Updated enum must:
+//   - Have at-most the same number of enum-cases as the old enum (Removing is allowed, but no additions).
```

Comment:
I think it's actually the other way round: 

Adding new cases is OK, because there are no stored values with the associated raw value. Accessing the new case on the enum type (e.g. `E.newCase`) won't be invalid like adding a new field to a composite, because the enum type value (`E`) is never stored, and is only created when the enum declaration is interpreted.

Removing a case is then actually not valid, because this still lead to confusion in a two-step process: First, remove a case in the first update, then add a new case in another update.

Reply:
oh yeah! we could do the other way around, that would also be less painful to the users.

Reply:
added in https://github.com/onflow/cadence/pull/762/commits/d056627a32781deb439492df48a9c8df3ff6003c

Reply:
@SupunS Wow, so fast! 👏